### DISCUSSION TITLE
request assistance resizing and centering svg

### DISCUSSION THREAD

**Discussion by ReenigneArcher** [spakin/SimpInkScr#138] [Q&A]
I need to resize and center ~230 svgs, I figured it would be better to automate this, but running into some issues.

This is what I need to do specifically.

1. resize all objects (as if they are one object) in page to a max dimension of 128 (without distorting opposite axis)... scaling the group doesn't seem to work...
2. resize canvas to 128 x 128
3. center new image (all objects) in the canvas

```py
try:
    from simpinkscr import *
except ImportError:
    pass

c = canvas

icon = all_shapes()

# initial resize so we can get the total width and height
c.resize_to_content()

# the final size needs to have one of the dimensions as 128
scale = min(128.0 / c.width, 128.0 / c.height)

for shape in icon:
    shape.scale(
        factor=scale,
    )

canvas.viewbox = (0, 0, 128, 128)
```

Issues I'm facing

1. The scale operation doesn't seem to work when there are multiple objects
2. The final line is not working as I expect (supposed to resize the "page"?) (this works though https://github.com/spakin/SimpInkScr/discussions/72#discussioncomment-4762241)
3. How to align the new image to the center of the page? Replicating the UI commands shown
![image](https://github.com/user-attachments/assets/ac2e9300-38ae-4863-8e59-dd2960a20d3e)

4. How can define the `canvas` object so my IDE doesn't complain about it? (and can give typing hints)

Any guidance would be appreciated. Thank you!

**Comment by ReenigneArcher** [comment]
I ended up manually resizing them. So this is not pressing, although I would still like to know how to do this for possible future cases.

**Reply by spakin** [reply]
Sorry to be unresponsive; I'm on travel all week and not near my Simple Inkscape Scripting development desktop.

My initial thought is to create a group out of `all_shapes()`, scale the group, rescale the canvas, center the group, and finally ungroup the shapes.  Perhaps you can try an approach like that?

**Reply by spakin** [reply]
To be more specific, see if the following Simple Inkscape Scripting code implements the transformation you had intended:

```Python
try:
    from simpinkscr import *
except ImportError:
    pass

c = canvas
icon = group(all_shapes())
bbox = icon.bounding_box()

# the final size needs to have the larger dimension as 128
scale = min(128.0 / bbox.width, 128.0 / bbox.height)
icon.scale(scale)
new_bbox = icon.bounding_box()

# move the icon to the upper-left corner of the canvas
ofs = ((128.0 - new_bbox.width)/2, (128.0 - new_bbox.height)/2)
icon.translate((ofs[0] - new_bbox.left, ofs[1] - new_bbox.top))

# resize the canvas to the icon
c.true_width = 128.0
c.true_height = 128.0
c.viewbox = [0, 0, 128, 128]

# restore the icon's individual shapes
icon.ungroup()
```
